app/models/spree/gift_card.rb

```diff
@@ -168,32 +170,33 @@ def can_capture?(payment)
 
     private
 
-    def redeem(user)
+    def redeem(user, current_store)
       begin
         transaction do
           previous_current_value = amount_remaining
           debit(amount_remaining)
-          build_store_credit(user, previous_current_value).save!
+          build_store_credit(user, previous_current_value, current_store).save!
         end
       rescue Exception => e
         self.errors[:base] = 'There some issue while redeeming the gift card.'
         false
       end
     end
 
-    def build_store_credit(user, previous_current_value)
+    def build_store_credit(user, previous_current_value, current_store)  
       user.store_credits.build(
             amount: previous_current_value,
             category: Spree::StoreCreditCategory.gift_card.last,
-            memo: "Gift Card - #{ variant.product.name } received from #{ recieved_from }",
+            memo: "Gift Card - #{ variant&.product&.name } received from #{ recieved_from }",
             created_by: user,
             action_originator: user,
-            currency: Spree::Config[:currency]
+            currency: Spree::Config[:currency],
+            store: current_store
         )
     end
 
     def recieved_from
-      line_item.order.email
+      self.created_by.eql?('admin') ? 'admin' : line_item.order.email 
```

Comment:
why we change this?

Reply:
The purpose of differentiating based on email is to determine whether each gift card originated from a user's email or an admin's email.